src/gpio/gpio.c

```diff
@@ -61,68 +62,107 @@ mraa_gpio_init_internal(mraa_adv_func_t* func_table, int pin)
         return NULL;
 
     mraa_result_t status = MRAA_SUCCESS;
-    char bu[MAX_SIZE];
-    int length;
+    mraa_board_t* board = plat;
+    struct stat dir_stat;
+    unsigned int phy_pin;
+    char bu[MAX_SIZE], pin_name[MRAA_PIN_NAME_SIZE];
+    int export_fd;
+    int length, i;
 
     mraa_gpio_context dev = (mraa_gpio_context) calloc(1, sizeof(struct _gpio));
     if (dev == NULL) {
         syslog(LOG_CRIT, "gpio%i: Failed to allocate memory for context", pin);
         return NULL;
     }
-
     dev->advance_func = func_table;
     dev->pin = pin;
-
+    dev->gpio_path = (char*) calloc(MAX_SIZE, sizeof(char));
+    if (dev->gpio_path == NULL) {
+        syslog(LOG_CRIT, "gpio%i: Failed to allocate memory for context's attribute", pin);
+        status = MRAA_ERROR_INVALID_RESOURCE;
+        goto init_internal_cleanup;
+    }
     if (IS_FUNC_DEFINED(dev, gpio_init_internal_replace)) {
         status = dev->advance_func->gpio_init_internal_replace(dev, pin);
         if (status == MRAA_SUCCESS)
             return dev;
         else
             goto init_internal_cleanup;
     }
-
     if (IS_FUNC_DEFINED(dev, gpio_init_pre)) {
         status = dev->advance_func->gpio_init_pre(pin);
         if (status != MRAA_SUCCESS)
             goto init_internal_cleanup;
     }
-
     dev->value_fp = -1;
     dev->isr_value_fp = -1;
 #ifndef HAVE_PTHREAD_CANCEL
     dev->isr_control_pipe[0] = dev->isr_control_pipe[1] = -1;
 #endif
     dev->isr_thread_terminating = 0;
     dev->phy_pin = -1;
+    dev->owner = 1;
+    snprintf(dev->gpio_path, MAX_SIZE, "/gpio%d", dev->pin);
+    snprintf(bu, sizeof(bu), SYSFS_CLASS_GPIO "%s", dev->gpio_path);
```

Comment:
It seems like `gpio_path` is empty at this point,just a freshly allocated and zeroed out piece of memory. What's the idea here?